lib/diagnostic.js

```diff
@@ -15,8 +17,11 @@ const min = function (previous, current) {
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let total = array.reduce(sum);`
 const sum = function (previous, current) {
+  let total = sum.reduce(function (prevous, current) {
```

Comment:
Yup. You didn't need to write out the entire function for this one however, simply the callback function that is passed to the `reduce`.